agenta-backend/agenta_backend/routers/app_router.py

```diff
@@ -594,8 +594,8 @@ async def create_app_and_variant_from_template(
 
             envvars = {**(payload.env_vars or {})}
             for key in supported_llm_prodviders_keys:
-                if os.environ.get(key):
-                    envvars[key] = os.environ[key]
+                if envvars.get(key):
+                    envvars[key] = envvars[key]
```

Comment:
This does not make sense

Reply:
So regardless, I should still include the logic that makes use of our API key if the user does not include theirs? Got it.